plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1935,12 +1932,7 @@ private void writeCheckpointIfNeeded(ConnectorSession session, SchemaTableName t
             checkpointWriterManager.writeCheckpoint(session, snapshot);
         }
         catch (Exception e) {
-            if (ignoreCheckpointWriteFailures) {
-                LOG.warn(e, "Failed to write checkpoint for table %s for version %s", table, newVersion);
-            }
-            else {
-                throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Failed to write checkpoint for table %s for version %s", table, newVersion), e);
-            }
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Failed to write checkpoint for table %s for version %s", table, newVersion), e);
```

Comment:
`writeCheckpointIfNeeded` is called after the write has been "committed", i.e. the transaction log exists.
Let's assume it's an INSERT query. If we now fail it, it can be retried by the calling application, leading to duplicate rows.

IMO we should log unconditionally 

@alexjo2144 wdyt?

Reply:
What loggin buys us? If so I would make exception message (for which it is more probable that someone would actually read it) more explanatory (that data could have been (or have been) already persisted in the table even though query failed).